File: DriftGame/uracer-desktop/src/com/bitfire/uracer/URacerDesktopFinalizer.java
package com.bitfire.uracer;

import com.badlogic.gdx.backends.lwjgl.audio.OpenALAudio;
import com.bitfire.uracer.URacer.URacerFinalizer;
import com.bitfire.uracer.configuration.BootConfig;

import org.lwjgl.opengl.Display;

public class URacerDesktopFinalizer implements URacerFinalizer {
    BootConfig boot = null;
    private OpenALAudio audio = null;

    public URacerDesktopFinalizer(BootConfig boot, OpenALAudio audio) {
        this.boot = boot;
        this.audio = audio;
    }

    @Override
    public void dispose() {
        boot.load();
        boot.setWindowX(Display.getX());
        boot.setWindowY(Display.getY());
        boot.store();

        // destroy display
        Display.destroy();

        // destroy audio, if any
        if (this.audio != null) {
            this.audio.dispose();
            this.audio = null;
        }
    }
}

File: DriftGame/uracer-desktop/src/com/bitfire/uracer/URacerDesktopFinalizer.kt
package com.bitfire.uracer

import com.badlogic.gdx.backends.lwjgl.audio.OpenALAudio
import com.bitfire.uracer.URacer.URacerFinalizer
import com.bitfire.uracer.configuration.BootConfig
import org.lwjgl.opengl.Display

class URacerDesktopFinalizer(val boot: BootConfig, private var audio: OpenALAudio?) : URacerFinalizer {

    override fun dispose() {
        boot.load()
        boot.windowX = Display.getX()
        boot.windowY = Display.getY()
        boot.store()

        // destroy display
        Display.destroy()

        // destroy audio, if any
        if (this.audio != null) {
            this.audio?.dispose()
            this.audio = null
        }
    }
}
